src/CLI/Config.php

```diff
@@ -11,14 +11,32 @@ class Config
 {
     /** @var array */
     private $classSetRules;
+    /** @var bool */
+    private $runOnlyARule;
 
     public function __construct()
     {
         $this->classSetRules = [];
+        $this->runOnlyARule = false;
     }
 
     public function add(ClassSet $classSet, ArchRule ...$rules): self
     {
+        if ($this->runOnlyARule) {
+            return $this;
+        }
+
+        /** @var ArchRule $rule */
+        foreach ($rules as $rule) {
+            if ($rule->isRunOnlyThis()) {
+                $rules = [];
+                $rules[] = $rule;
+
```

Comment:
maybe it's simpler to insert the return here and remove the runOnlyARule attribute?

Reply:
You can call multiple times this method with different ClassSet, it's not the best approach I know but adding the `return` here I think it's not correct.
What do you think?

Reply:
oh, i get it now... you are right!